spec/formatting.md

```diff
@@ -18,3 +18,236 @@ the local variable takes precedence.
+    does not provide for the variable reference `$var` to be successfully resolved:
+
+    ```
+    {The value is {$var}.}
+    ```
+
+    ```
+    match {$var}
+    when 1 {The value is one.}
+    when * {The value is not one.}
+    ```
+
+  - **Unknown Function errors** occur when an Expression includes
+    a reference to a function which cannot be resolved.
+
+    For example, attempting to format either of the following messages
+    must result in an Unknown Function error if done within a context that
+    does not provide for the function `:func` to be successfully resolved:
+
+    ```
+    {The value is {(horse) :func}.}
+    ```
+
+    ```
+    match {(horse) :func}
+    when 1 {The value is one.}
+    when * {The value is not one.}
+    ```
+
+- **Selection errors** occur when message selection fails.
+
+  - **Selector errors** are failures in the matching of a key to a specific selector.
+
+    For example, attempting to format either of the following messages
+    may result in a Selector error if done within a context that
+    uses a `:plural` selector function which requires its input to be numeric:
+
+    ```
+    match {(horse) :plural}
+    when 1 {The value is one.}
+    when * {The value is not one.}
+    ```
+
+    ```
+    let $sel = {(horse) :plural}
+    match {$sel}
+    when 1 {The value is one.}
+    when * {The value is not one.}
+    ```
+
+- **Formatting errors** occur during the formatting of a resolved value,
+  for example when encountering a value with an unsupported type
+  or an internally inconsistent set of options.
+
+  For example, attempting to format any of the following messages
+  may result in a Formatting error if done within a context that
+
+  1. provides for the variable reference `$user` to resolve to
+     an object `{ name: 'Kat', id: 1234 }`, and
+  2. uses a `:get` formatting function which requires its argument to be an object and
+     an option `field` to be provided with a string value,
+
+  ```
+  {Hello, {(horse) :get field=name}!}
+  ```
+
+  ```
+  {Hello, {$user :get}!}
+  ```
+
+  ```
+  let $id = {$user :get field=id}
+  {Hello, {$id :get field=name}!}
+  ```
+
+Syntax and Data Model errors must be emitted as soon as possible.
+
+During selection, an Expression handler must only emit Resolution and Selection errors.
+During formatting, an Expression handler must only emit Resolution and Formatting errors.
+
+In all cases, when encountering an error,
+a message formatter must provide some representation of the message.
+An informative error or errors must also be separately provided.
+When a message contains more than one error,
+or contains some error which leads to further errors,
+an implementation which does not emit all of the errors
+should prioritise Syntax and Data Model errors over others.
+
+When an error occurs in the resolution of an Expression or Markup Option,
+the Expression or Markup in question is processed as if the option were not defined.
```

Comment:
as if the option **was** not defined

Reply:
As @macchiati noted, this is correct English as-is.